runner/models.py

```diff
@@ -48,6 +95,27 @@ class Run(BaseModel):
     job_statuses = JSONField(default=dict, blank=True)
     output_metadata = JSONField(default=dict, blank=True, null=True)
     tags = JSONField(default=dict, blank=True, null=True)
+    operator_run = models.ForeignKey(OperatorRun, on_delete=models.CASCADE, null=True, related_name="runs")
+
+    def __init__(self, *args, **kwargs):
+        super(Run, self).__init__(*args, **kwargs)
+        # TODO change state can be handled with a mixin
+        self.original = {
+            "status": self.status
+        }
+
+    def save(self, *args, **kwargs):
+        # TODO do we want to decrement if a job goes from completed/failed to open or failed to complete?
+        # We can also a prevent a job from going to open once it's in a closed state
+        if self.operator_run and self.original["status"] != self.status:
+            if self.status == RunStatus.COMPLETED:
+                self.operator_run.increment_completed_run()
+                self.original["status"] = RunStatus.COMPLETED
```

Comment:
I am not sure what is `self.original`?

Reply:
It's for this condition here:
https://github.com/mskcc/beagle/pull/146/files/c027c32edba314f9fa800e4090870deb92ade977#diff-e27fa3dbcada827773bbfa1157890a70R110

Let's say we update the status of a run:
```
run.status = RunStatus.COMPLETED
```

There's no way, afaik, to check whether the `status` property was changed.